pkg/services/ngalert/state/template_test.go

```diff
@@ -0,0 +1,392 @@
+			},
+		},
+		expected: "100ms:100us:123.5ms:1m 0s:1m 0s:1.234s:12.35s:",
+	}, {
+		name: "humanizeDuration - subsecond - string",
+		text: "{{ humanizeDuration $value }}",
+		alertInstance: eval.Result{
+			EvaluationString: ".0001",
+		},
+		expected: "100us",
+	}, {
+		name: "humanizeDuration - fractional seconds - string",
+		text: "{{ humanizeDuration $value }}",
+		alertInstance: eval.Result{
+			EvaluationString: "1.2345",
+		},
+		expected: "1.234s",
+	}, {
+		name: "humanizeDuration - string with error",
+		text: `{{ humanizeDuration $value }}`,
+		alertInstance: eval.Result{
+			EvaluationString: "invalid",
+		},
+		expectedError: errors.New(`error executing template __alert_test: template: __alert_test:1:79: executing "__alert_test" at <humanizeDuration $value>: error calling humanizeDuration: strconv.ParseFloat: parsing "invalid": invalid syntax`),
+	}, {
+		name:     "humanizePercentage - float64",
+		text:     "{{ -0.22222 | humanizePercentage }}:{{ 0.0 | humanizePercentage }}:{{ 0.1234567 | humanizePercentage }}:{{ 1.23456 | humanizePercentage }}",
+		expected: "-22.22%:0%:12.35%:123.5%",
+	}, {
+		name:     "humanizePercentage - string",
+		text:     `{{ "-0.22222" | humanizePercentage }}:{{ "0.0" | humanizePercentage }}:{{ "0.1234567" | humanizePercentage }}:{{ "1.23456" | humanizePercentage }}`,
+		expected: "-22.22%:0%:12.35%:123.5%",
+	}, {
+		name:          "humanizePercentage - string with error",
+		text:          `{{ "invalid" | humanizePercentage }}`,
+		expectedError: errors.New(`error executing template __alert_test: template: __alert_test:1:91: executing "__alert_test" at <humanizePercentage>: error calling humanizePercentage: strconv.ParseFloat: parsing "invalid": invalid syntax`),
+	}, {
+		name:     "humanizeTimestamp - float64",
+		text:     "{{ 1435065584.128 | humanizeTimestamp }}",
+		expected: "2015-06-23 13:19:44.128 +0000 UTC",
+	}, {
+		name:     "humanizeTimestamp - string",
+		text:     `{{ "1435065584.128" | humanizeTimestamp }}`,
+		expected: "2015-06-23 13:19:44.128 +0000 UTC",
+	}, {
+		name:     "title",
+		text:     `{{ "aa bb CC" | title }}`,
+		expected: "Aa Bb CC",
+	}, {
+		name:     "toUpper",
+		text:     `{{ "aa bb CC" | toUpper }}`,
+		expected: "AA BB CC",
+	}, {
+		name:     "toLower",
+		text:     `{{ "aA bB CC" | toLower }}`,
+		expected: "aa bb cc",
+	}, {
+		name:     "match",
+		text:     `{{ match "a+" "aa" }} {{ match "a+" "b" }}`,
+		expected: "true false",
+	}, {
+		name:     "match",
+		text:     `{{ match "a+" "aa" }} {{ match "a+" "b" }}`,
+		expected: "true false",
+	}, {
+		name:     "regex replacement",
+		text:     "{{ reReplaceAll \"(a)b\" \"x$1\" \"ab\" }}",
+		expected: "xa",
+	}, {
+		name:     "pass multiple arguments to templates",
+		text:     `{{define "x"}}{{.arg0}} {{.arg1}}{{end}}{{template "x" (args 1 "2")}}`,
+		expected: "1 2",
+	}, {
+		name:     "pathPrefix",
+		text:     "{{ pathPrefix }}",
+		expected: pathPrefix,
+	}, {
+		name:     "externalURL",
```

Comment:
Ditto from my message above. Add it to the documentation and then document it as part of #39662